src/windows/disk.rs

```diff
@@ -245,18 +333,29 @@ pub(crate) unsafe fn get_list() -> Vec<Disk> {
                     DiskKind::SSD
                 }
             };
-            Some(Disk {
-                inner: DiskInner {
-                    type_,
-                    name: name.to_owned(),
-                    file_system: file_system.to_vec(),
-                    mount_point: mount_point.to_vec(),
-                    s_mount_point: String::from_utf16_lossy(&mount_point[..mount_point.len() - 1]),
-                    total_space,
-                    available_space,
-                    is_removable,
-                },
-            })
+
+            let name_len = name.iter().position(|&x| x == 0).unwrap_or(name.len());
+            let name = OsString::from_wide(&name[..name_len]);
+            let file_system = file_system
+                .iter()
+                .take_while(|c| **c != 0)
+                .map(|c| *c as u8)
```

Comment:
Is there a case where a `u16` could be needed here?

Reply:
You tell me 😉! The public [`Disk` interface](https://github.com/GuillaumeGomez/sysinfo/blob/ade89e39c26599db39f2a4a847e79908fbab1154/src/common.rs#L1925C5-L1925C39) exposes a `&[u8]`. The [original code](https://github.com/GuillaumeGomez/sysinfo/blob/ade89e39c26599db39f2a4a847e79908fbab1154/src/windows/disk.rs#L201C43-L201C94) also truncates the code units to `u8` 😢 

My gut reaction is that I would make the `Disk` interface expose a `&OsStr` instead, but that seems like a big hassle (breaking change) for relatively little benefit.

Reply:
Next release will be a major one as I already completely changed the API. If you don't mind changing that (either in this PR or in a follow-up), I'll gladly accept it.

Reply:
Happy to do the `[u8]` -> `OsStr` change, but I'd prefer to do it in a separate PR.

Reply:
I'd prefer too so let's go for another PR.